src/clients/db/Repository.ts

```diff
@@ -41,6 +43,35 @@ export default class Repository<T extends AbstractDBModel> {
     return result.docs.map(mapDocument)
   }
 
+  async findAllPaged(sort = Unsorted, pageRequest: PageRequest = UnpagedRequest): Promise<Page<T>> {
+    const selector: any = {
+      _id: { $gt: null },
+    }
+
+    sort.sorts.forEach((s) => {
+      selector[s.field] = { $gt: null }
+    })
+
+    const result = await this.db.find({
+      selector,
+      sort: sort.sorts.length > 0 ? sort.sorts.map((s) => ({ [s.field]: s.direction })) : undefined,
+      limit: pageRequest.limit,
+      skip: pageRequest.skip,
```

Comment:
In PouchDB, skip is very slow. It literally is doing what it says, "skips" the elements that we don't care about. This means that the elements that are getting "skipped" are still loaded in to memory. 

Checkout this blog post: https://pouchdb.com/2014/04/14/pagination-strategies-with-pouchdb.html.

it provides a mechanism for "smart" skip.

Reply:
To accomplish this, our PageRequest may need to know the previous start key. However there could be a variety of implementations.

Reply:
Agreed that skip is an expensive operation. I intend to add index on `fullName`, run query on the index. https://pouchdb.com/guides/mango-queries.html#pagination.
But, I am facing issue in pagination:
1. `fullname` cannot be unique (thus needs to be coupled with other field which is a sequence)
2. As `code` or `_id` field is not a sequence, the pagination logic will fail.
currently,  assume records : ``[...{fullName: def, code: P-zzz}, {fullName: ghi, code: P-aaa}...]``
and if last record in a page is `{fullName: def, code: P-zzz}` on clicking next page,
query will be: find docs with `fullName > abc`  and `code > P-zzz`.
Query will not return `{fullName: ghi, code: P-aaa}`.
I propose to change `code` to a sequence or add new field that will be sequence(redundant).

Reply:
I think index on `[fullName, _id]` will work.

Reply:
I tried with `[fullName, _id]` but did not work. The best solution I can think of is add new field `index` which will be `fullName + code` and thus make sure uniqueness along with indexing.